tests/functional/api.php

```diff
@@ -0,0 +1,57 @@
+<?php
+define('INSTALL_ROOT',  realpath(__DIR__ . '/../../') . '/');
+
+// We always work with UTF8 encoding
+mb_internal_encoding('UTF-8');
+
+// Make sure we have a timezone set
+date_default_timezone_set('Europe/Paris');
+
+require __DIR__ . '/../../vendor/autoload.php';
+
+// Launch PHP dev server in the background
+chdir(INSTALL_ROOT);
+exec('php -S 0.0.0.0:8083 > /dev/null 2>&1 & echo $!', $output);
+
+// We will need the pid to kill it, beware, this is the pid of the bash process started above
+$pid = $output[0];
+
+// Pause to let time for the dev server to launch in the background
+sleep(3);
+
+$paths = [
+    ['?action=listlocales&json', 400, "[\n    \"Please check you request: provide a project name, or a valid couple website+file.\"\n]"],
+    ['?website=0&file=firefox/desktop/tips.lang&json&locale=fr', 200, "{\n    \"firefox\/desktop\/tips.lang\": {\n        \"fr\": {\n            \"Identical\": 0,\n            \"Translated\": 40,\n            \"Missing\": 0,\n            \"Obsolete\": 0,\n            \"tags\": [\n                \"sync_device_note\"\n            ],\n            \"activated\": true\n        }\n    }\n}"],
+    ['?website=foo&file=main.lang&json', 400, 'foo is not a supported website. Check the value and try again.'],
```

Comment:
Yes, we have several of them. Either you duplicate `$check` and avoid doing `->isJson()` for these calls /ì(haven't tried but it should work), or update the code to always return a JSON.